compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1049,6 +1050,10 @@ object Types {
       case _ => this
     }
 
+    def stripLazyRef(given Context): Type = this match
+      case lzy: LazyRef => lzy.ref
```

Comment:
Maybe recursively call `stripLazyRef` on the result ? This would match the behavior of other `strip...` methods on Type:
```suggestion
      case lzy: LazyRef => lzy.ref.stripLazyRef
```